llama.cpp

```diff
@@ -7973,23 +7973,26 @@ void llama_sample_softmax(struct llama_context * ctx, llama_token_data_array * c
 }
 
 void llama_sample_top_k(struct llama_context * ctx, llama_token_data_array * candidates, int32_t k, size_t min_keep) {
-    const int64_t t_start_sample_us = ggml_time_us();
+    if (candidates->sorted) {
+       candidates->size = k;
+       return;
+    }
 
+    const int64_t t_start_sample_us = ggml_time_us();
+    
     k = std::max(k, (int) min_keep);
     k = std::min(k, (int) candidates->size);
```

Comment:
I'm realizing now that these sanity checks will not run if we return early. Does that matter?

Reply:
I think they need to remain - simply move them before the "if sorted" early return at the top